File: 0205-isomorphic-strings/0205-isomorphic-strings.java
class Solution {
    public boolean isIsomorphic(String s, String t) {
        int n = s.length();
        Map<Character , Character> map = new HashMap<Character , Character>();
        Map<Character , Character> map2 = new HashMap<Character , Character>();
        for(int i = 0 ; i < n ; i++){
            char x = s.charAt(i);
            char y = t.charAt(i);
            if(map.containsKey(x) && map.get(x) != y) return false;
            if(map2.containsKey(y) && map2.get(y) != x) return false;
            map.put(x , y);
            map2.put(y , x);
        }        
        return true;
    }
}

File: 0205-isomorphic-strings/0205-isomorphic-strings.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        hashmap_s, hashmap_t = {}, {}
        for c, d in zip(s, t):
            if d in hashmap_t and hashmap_t[d] != c:
                return False
            if c in hashmap_s and hashmap_s[c] != d:
                return False
            hashmap_s[c], hashmap_t[d] = d, c
        return True